RHEL8 Support (for dnf-automatic)

jadestorm opened the pull request:
Howdy!  This PR adds support for RHEL8 by adding support for DNF.  I originally was thinking about doing a separate module for dnf-automatic, but it's so similar to yum-cron and also is really just Yum 4 that it seemed like it made more sense to keep it here.

Now -- I'm submitting the PR to start a conversation atm.  I haven't added any testing or documentation components yet.  Could you please take a look and let me know what you think and if you are happy with this solution I'll polish things up?  Thanks!

This is in reference to #21

treydock commented on manifests/config.pp:
```diff
@@ -6,7 +6,25 @@
+      dnf_automatic_config { 'email/email_to': value => $yum_cron::mailto }
+      dnf_automatic_config { 'email/email_host': value => $yum_cron::email_host }
+      dnf_automatic_config { 'base/debuglevel': value => $yum_cron::debug_level }
+
```
Travis lint tests failing here, unnecessary white space?

treydock commented on manifests/config.pp:
```diff
@@ -6,7 +6,25 @@
+
+      create_resources(dnf_automatic_config, $yum_cron::extra_configs)  
+    }
+  
```
Travis lint tests failing here, unnecessary white space?

jadestorm commented:
Howdy @treydock -- Interesting, fixed the whitespace.  While I was at it I went ahead and updated the README and metadata.json and consolidated my commits.  that said, I'm not sure how to approach fixing the spec tests at the moment.  There's certainly quite a few adjustments that need to be made to make those pass.  Any thoughts?  Would you prefer to take that on yourself or me do it?  If so, can you give me some general guidance on how you'd like to see it worked in?

treydock commented:
Looks good. I won't be able to release until I convert this module to using Litmus instead of Beaker for acceptance tests as Beaker and Docker has issues for CentOS 8 containers.

treydock commented:
Released in v5.1.0
